File: backend/olympic_it_project/src/main/java/org/example/olympic_ot_project/dto/exam/question/QuestionPageResponse.java
package org.example.olympic_ot_project.dto.exam.question;

import lombok.Data;

import java.util.List;

@Data
public class QuestionPageResponse {

    private List<QuestionResponse> items;

    private int page;
    private int size;
    private long totalElements;
    private int totalPages;
}

File: frontend/olympic_it_project/olympic_it_project/lib/dto/admin_manager/question/question_page_response.dart
import 'package:olympic_it_project/dto/admin_manager/question/question_response.dart';

class QuestionPageResponse {
  final List<QuestionResponse> items;
  final int page;
  final int size;
  final int totalElements;
  final int totalPages;

  QuestionPageResponse({
    required this.items,
    required this.page,
    required this.size,
    required this.totalElements,
    required this.totalPages,
  });

  factory QuestionPageResponse.fromJson(Map<String, dynamic> json) {
    return QuestionPageResponse(
      items: (json["items"] as List)
          .map((e) => QuestionResponse.fromJson(e))
          .toList(),
      page: json["page"],
      size: json["size"],
      totalElements: json["totalElements"],
      totalPages: json["totalPages"],
    );
  }
}